src/components/SmartLoan.vue

```diff
@@ -0,0 +1,327 @@
+            <img src="src/assets/icons/slash-small.svg"/>
+            <img @click="showCollateralBlock(1)" src="src/assets/icons/minus.svg" class="minus"/>
+          </div>
+      </div>
+    </Bar>
+    <Block v-if="borrowBlock" class="block borrow-block" :bordered="true">
+      <img @click="borrowBlock = false" src="src/assets/icons/cross.svg" class="cross" />
+      <Tabs :openTabIndex="tabIndex">
+        <Tab title="Borrow" imgActive="add-deposit-active" img="add-deposit" imgPosition="left">
+          <CurrencyForm
+            label="Borrow"
+            v-on:submitValue="borrowValue"
+            :waiting="waitingForBorrow"
+            flexDirection="column"
+            :style="{'width': '490px'}"
+            :validators="[
+              {require: value => value <= getAvailable, message: 'Borrow amount exceeds amount available in the pool'},
+              {require: value => calculateSolvency(0, value) >= 1.2, message: 'New solvency ratio is below acceptable level'},
+            ]"
+            :info="value => `New solvency ratio: <b>${(calculateSolvency(0, value) * 100).toFixed(1)}%</b>`"
+          />
+        </Tab>
+        <Tab title="Repay" imgActive="withdraw-deposit-active" img="withdraw-deposit" imgPosition="right">
+          <CurrencyForm
+            label="Repay"
+            v-on:submitValue="repayValue"
+            :waiting="waitingForRepay"
+            flexDirection="column"
+            :style="{'width': '490px'}"
+            :validators="[
+              {require: (value) => value <= debt, message: 'Repay amount exceeds borrowed amount'}
+            ]"
+            :info="value => `New solvency ratio: <b>${(calculateSolvency(0, -value) * 100).toFixed(1)}%</b>`"
+          />
+        </Tab>
+      </Tabs>
+    </Block>
+    <Block v-if="collateralBlock" class="block collateral-block" :bordered="true">
+      <img @click="collateralBlock = false" src="src/assets/icons/cross.svg" class="cross" />
+      <Tabs :openTabIndex="tabIndex">
+        <Tab title="Add collateral" imgActive="add-deposit-active" img="add-deposit" imgPosition="left">
+          <CurrencyForm
+            label="Add"
+            v-on:submitValue="fundValue"
+            :waiting="waitingForFund"
+            flexDirection="column"
+            :style="{'width': '490px'}"
+            :validators="[
+              {require: (value) => value <= balance, message: 'Repay amount exceeds user balance'}
+            ]"
+            :info="value => `New solvency ratio: <b>${(calculateSolvency(value, 0) * 100).toFixed(1)}%</b>`"
+          />
+        </Tab>
+        <Tab title="Reduce collateral" imgActive="withdraw-deposit-active" img="withdraw-deposit" imgPosition="right">
+          <CurrencyForm
+            label="Reduce"
+            v-on:submitValue="withdrawValue"
+            :waiting="waitingForWithdraw"
+            flexDirection="column"
+            :style="{'width': '490px'}"
+            :validators="[
```

Comment:
I would move the :validators and :info values to the script part of the component